test/helper/common.bash

```diff
@@ -1,247 +1,397 @@
-  # unless explicitly set (useful for waiting on a min count to be reached):
-  if [[ -z $EXPECTED_COUNT ]]
-  then
-    # +1 of starting count:
-    EXPECTED_COUNT=$(( $(__get_count) + 1 ))
-  fi
+  _run_in_container setup email add "${MAIL_ACCOUNT}" "${MAIL_PASS}"
+  assert_success
 
-  repeat_until_success_or_timeout 20 __has_expected_count
+  _wait_until_account_maildir_exists "${MAIL_ACCOUNT}"
 }
 
-# An account added to `postfix-accounts.cf` must wait for the `changedetector` service
-# to process the update before Dovecot creates the mail account and associated storage dir:
-function wait_until_account_maildir_exists() {
-  local CONTAINER_NAME=$1
-  local MAIL_ACCOUNT=$2
+# Assert that the number of lines output by a previous command matches the given
+# amount (${1}). `lines` is a special BATS variable updated via `run`.
+#
+# @param ${1} = number of lines that the output should have
+function _should_output_number_of_lines() {
+  assert_equal "${#lines[@]}" "${1:?Number of lines not provided}"
+}
 
-  local LOCAL_PART="${MAIL_ACCOUNT%@*}"
-  local DOMAIN_PART="${MAIL_ACCOUNT#*@}"
-  local MAIL_ACCOUNT_STORAGE_DIR="/var/mail/${DOMAIN_PART}/${LOCAL_PART}"
+# Reloads the postfix service.
+#
+# @param ${1} = container name [OPTIONAL]
+function _reload_postfix() {
+  local CONTAINER_NAME=$(__handle_container_name "${1:-}")
 
-  repeat_in_container_until_success_or_timeout 60 "${CONTAINER_NAME}" bash -c "[[ -d ${MAIL_ACCOUNT_STORAGE_DIR} ]]"
+  # Reloading Postfix config after modifying it within 2 seconds will cause Postfix to delay reading `main.cf`:
+  # WORKAROUND: https://github.com/docker-mailserver/docker-mailserver/pull/2998
+  _exec_in_container touch -d '2 seconds ago' /etc/postfix/main.cf
+  _exec_in_container postfix reload
 }
 
-function add_mail_account_then_wait_until_ready() {
-  local CONTAINER_NAME=$1
-  local MAIL_ACCOUNT=$2
-  # Password is optional (omit when the password is not needed during the test)
-  local MAIL_PASS="${3:-password_not_relevant_to_test}"
 
-  run docker exec "${CONTAINER_NAME}" setup email add "${MAIL_ACCOUNT}" "${MAIL_PASS}"
-  assert_success
+# Get the IP of the container (${1}).
+#
+# @param ${1} = container name [OPTIONAL]
+function _get_container_ip() {
+  local TARGET_CONTAINER_NAME=$(__handle_container_name "${1:-}")
+  docker inspect --format '{{ .NetworkSettings.IPAddress }}' "${TARGET_CONTAINER_NAME}"
+}
 
-  wait_until_account_maildir_exists "${CONTAINER_NAME}" "${MAIL_ACCOUNT}"
+# Check if a container is running.
+#
+# @param ${1} = container name [OPTIONAL]
+function _container_is_running() {
+  local TARGET_CONTAINER_NAME=$(__handle_container_name "${1:-}")
+  [[ $(docker inspect -f '{{.State.Running}}' "${TARGET_CONTAINER_NAME}") == 'true' ]]
 }
 
-function wait_for_empty_mail_queue_in_container() {
-  local CONTAINER_NAME="${1}"
-  local TIMEOUT=${TEST_TIMEOUT_IN_SECONDS}
+# Checks whether the directory exists and then how many files it container.
```

Comment:
Not sure if top-level is the right way to phrase it, but better to clarify the file check isn't recursive depth :sweat_smile: 

```suggestion
# Checks if the directory exists and then how many files it contains at the top-level.
```